site/src/api/queries/appearance.ts

```diff
@@ -0,0 +1,33 @@
+import { QueryClient } from "@tanstack/react-query";
+import * as API from "api/api";
+import { AppearanceConfig } from "api/typesGenerated";
+
+export const appearance = () => {
+  return {
+    queryKey: ["appearance"],
+    queryFn: fetchAppearance,
+  };
+};
+
+export const updateAppearance = (queryClient: QueryClient) => {
+  return {
+    mutationFn: API.updateAppearance,
+    onSuccess: (newConfig: AppearanceConfig) => {
+      queryClient.setQueryData(["appearance"], newConfig);
+    },
+  };
+};
+
+const fetchAppearance = () => {
+  const appearance = document.querySelector("meta[property=appearance]");
+  if (appearance) {
+    const rawContent = appearance.getAttribute("content");
+    try {
+      return JSON.parse(rawContent as string);
+    } catch (ex) {
+      // Ignore this and fetch as normal!
+    }
+  }
+
+  return API.getAppearance();
+};
```

Comment:
I feel like this function could get confused for a query easily. can we just inline it into `appearance` or stow it somewhere else?

Reply:
I created a function `getMetadataAsJSON` to abstract this a bit.